cli/crates/cli/tests/openapi/headers.rs

```diff
@@ -0,0 +1,100 @@
+//! Tests of passing through headers
+
+use std::{collections::BTreeMap, net::SocketAddr};
+
+use serde_json::Value;
+use wiremock::{
+    matchers::{method, path},
+    Mock, ResponseTemplate,
+};
+
+use crate::{http_spy::HttpSpy, utils::environment::Environment};
+
+use super::{doggie, mount_petstore_spec, start_grafbase};
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_header_passthrough() {
+    let mock_server = wiremock::MockServer::start().await;
+    mount_petstore_spec(&mock_server).await;
+
+    let mut env = Environment::init_async().await;
+    let client = start_grafbase(&mut env, petstore_schema_with_header_forwarding(mock_server.address())).await;
+
+    let http_spy = HttpSpy::new();
+
+    Mock::given(method("GET"))
+        .and(path("/pet/123"))
+        .and(http_spy.clone())
```

Comment:
Wasn't this enough?
https://docs.rs/wiremock/latest/wiremock/struct.MockServer.html#method.received_requests